test/mlir/krnl/krnl_shape_lowering.mlir

```diff
@@ -5,23 +5,21 @@ func @test_krnl_shape_lowering(%arg0: memref<?x?xf32>) -> index {
   %c0 = constant 0 : index
   %0 = dim %arg0, %c0 : memref<?x?xf32>
   %1 = alloc(%0) : memref<?x10xf32>
-  %shape = "krnl.shape"(%1) : (memref<?x10xf32>) -> !shape.shape
-  %d1 = "shape.get_extent"(%shape, %c1) : (!shape.shape, index) -> !shape.size
-  %ind = "shape.size_to_index"(%d1) : (!shape.size) -> index
-  %e = addi %ind, %ind : index
+  %shape = "krnl.shape"(%1) : (memref<?x10xf32>) -> memref<2xindex>
+  %e = load %shape[%c0] : memref<2xindex>
   return %e : index
 
   // CHECK-LABEL: test_krnl_shape_lowering
-  // CHECK: [[CONST0:%.+]] = constant 0 : index
-  // CHECK: [[CONST1:%.+]] = constant 1 : index
-  // CHECK: [[DIM:%.+]] = dim %arg0, [[CONST0]] : memref<?x?xf32>
+  // CHECK: %[[CONST0:.+]] = constant 0 : index
```

Comment:
are these changes because of moving the % outside the var name necessary? Makes it hard to see the actual change

Reply:
Yes it's a normal consequence of having that variable used inside square brackets in the code.